applications/GeoMechanicsApplication/custom_elements/steady_state_Pw_piping_element.cpp

```diff
@@ -312,26 +313,22 @@ double SteadyStatePwPipingElement<TDim,TNumNodes>:: CalculateEquilibriumPipeHeig
     const double SolidDensity = Prop[DENSITY_SOLID];
     const double FluidDensity = Prop[DENSITY_WATER];
  
-    // calculate pressure gradient over element
-    double dpdx = CalculateWaterPressureGradient(Prop, Geom, pipe_length);
+    // calculate head gradient over element
+    double dhdx = CalculateHeadGradient(Prop, Geom, pipe_length);
 
     // calculate particle diameter
     double particle_d = CalculateParticleDiameter(Prop);
     
     // todo calculate slope of pipe, currently pipe is assumed to be horizontal
     const double pipeSlope = 0;
 
-    // return infinite when dpdx is 0
-    if (dpdx < std::numeric_limits<double>::epsilon())
+    // return infinite when dhdx is 0
+    if (dhdx < std::numeric_limits<double>::epsilon())
     { 
         return 1e10;
     }
-
-    // gravity is taken from first node
-    array_1d<double, 3> gravity_array= Geom[0].FastGetSolutionStepValue(VOLUME_ACCELERATION);
-    const double gravity = norm_2(gravity_array);
 	
-    return modelFactor * M_PI / 3.0 * particle_d * (SolidDensity - FluidDensity) * gravity * eta  * sin((theta  + pipeSlope) * M_PI / 180.0) / cos(theta * M_PI / 180.0) / dpdx;
+    return modelFactor * M_PI / 3.0 * particle_d * (SolidDensity / FluidDensity - 1) * eta  * sin((theta  + pipeSlope) * M_PI / 180.0) / cos(theta * M_PI / 180.0) / dhdx;
```

Comment:
For a follow-up change we may consider to add a function that converts an angle in degrees to an angle in radians, and then use that function here (and probably in other places, too).

Reply:
For a new PR, I'll add issue on board